src/OpenTelemetry/BaseProcessor.cs

```diff
@@ -27,6 +27,18 @@ public BaseProcessor()
     /// </summary>
     public BaseProvider? ParentProvider { get; private set; }
 
+    /// <summary>
+    /// Gets or sets the weight of the processor when added to the provider
+    /// pipeline. Default value: <c>0</c>.
+    /// </summary>
+    /// <remarks>
+    /// Note: Weight is used to order processors when building a provider
+    /// pipeline. Lower weighted processors come before higher weighted
+    /// processors. Changing the weight after a pipeline has been constructed
+    /// has no effect.
+    /// </remarks>
+    internal int PipelineWeight { get; set; }
```

Comment:
I haven't spent time thinking about a proper name, but "pipeline weight" sounds wrong to me. Not a blocker though.